### DISCUSSION TITLE
"A newer package version is available in a configured source, but it does not apply to your system or requirements."

### DISCUSSION THREAD

**Discussion by MakotoE** [microsoft/winget-cli#4316] [Q&A] [answered]
I am trying to update a package, when the following output appears.

```
winget update --verbose-logs emurasoft.emeditor
No applicable upgrade found.
A newer package version is available in a configured source, but it does not apply to your system or requirements.
```

Note that I am trying to update from version `23.1.1` to the latest version, `24.0.1`. It will probably be reproducible on your machine with the following commands:

```
winget install emurasoft.emeditor -v 23.1.1
winget update --verbose-logs emurasoft.emeditor
```

My log for the update command includes this:

```
2024-03-26 19:15:53.600 [CLI ] Installer [X64,msi,User,en-US] not applicable: Installer scope does not match currently installed scope: User != Machine
2024-03-26 19:15:53.600 [CLI ] Installer [X86,msi,User,en-US] not applicable: Installer scope does not match currently installed scope: User != Machine
```

This is strange because we have only released user scope packages, never machine scope. This is confirmed for both versions with the command `winget list --scope user`.

The same issue happens when updating from `24.0.0` to `24.0.1`. A workaround for this issue is to uninstall the old version of emeditor and install the newer one.

[Here's the manifest directory for emeditor.](https://github.com/microsoft/winget-pkgs/tree/master/manifests/e/Emurasoft/EmEditor/24.0.1)

Any help with this issue is appreciated.

**Comment by mdanish-kh** [comment, marked as answer]
Should be resolved by 
- https://github.com/microsoft/winget-pkgs/pull/146930

**Reply by MakotoE** [reply]
Thank you. I confirm that the new version installs as expected.

**Reply by demo-err** [reply]
i encounter the same error

```
winget update ZedIndustries.Zed
No applicable upgrade found.
A newer package version is available in a configured source, but it does not app
ly to your system or requirements.

```

**Comment by johanvanmol** [comment]
I get the same error when trying to upgrade WinDirStat:

Related version:
```
Name                                                     Id                          Version       Available     Source
WinDirStat 1.1.2                                         WinDirStat.WinDirStat       < 2.0.1       2.0.1         winget
```

Command and output:
```
winget upgrade WinDirStat.WinDirStat
No applicable upgrade found.
A newer package version is available in a configured source, but it does not apply to your system or requirements.
```

Is there anything I can solve myself?  Or should this be fixed by Microsoft or the developer of WinDirStat?

**Reply by Masamune3210** [reply]
.....that isn't even close to the same error. Your's isn't even an error, for a start. 

You currently have installed a version which reports itself as either equal to or higher than what is in the manifest. There is nothing to do.

**Reply by larsch** [reply]
> .....that isn't even close to the same error. Your's isn't even an error, for a start.
> 
> You currently have installed a version which reports itself as either equal to or higher than what is in the manifest. There is nothing to do.

It says WinDirStat 1.1.2 is installed, which is less than 2.0.1.

I get the same error message with WinDirStat and Qalculate.Qalculate (5.2.0 => 5.3.0).

```
C:\>winget update --verbose WinDirStat.WinDirStat
No applicable upgrade found.
A newer package version is available in a configured source, but it does not apply to your system or requirements.
```

```
C:\Users\larsch>winget upgrade Qalculate.Qalculate
No applicable upgrade found.
A newer package version is available in a configured source, but it does not apply to your system or requirements.
```

For Qalculate, removing and reinstalling works just fine, but winget doesn't not want to upgrade it.

```
C:\Users\larsch>winget upgrade --all
Name                    Id                     Version Available Source
-----------------------------------------------------------------------
IrfanView 4.67 (64-bit) IrfanSkiljan.IrfanView 4.67    4.70      winget
Qalculate!              Qalculate.Qalculate    5.2.0   5.3.0     winget
VeraCrypt               IDRIX.VeraCrypt        1.25.9  1.26.15   winget
WinDirStat 1.1.2        WinDirStat.WinDirStat  < 2.0.1 2.0.1     winget
4 upgrades available.
2 package(s) have version numbers that cannot be determined. Use --include-unknown to see all results.
3 package(s) have pins that prevent upgrade. Use the 'winget pin' command to view and edit pins. Using the --include-pinned argument may show more results.
113 package(s) have version numbers that cannot be determined. Use --include-unknown to see all results.

C:\Users\larsch>winget upgrade Qalculate.Qalculate
No applicable upgrade found.
A newer package version is available in a configured source, but it does not apply to your system or requirements.

C:\Users\larsch>winget remove Qalculate.Qalculate
Found Qalculate! [Qalculate.Qalculate]
Starting package uninstall...
Successfully uninstalled

C:\Users\larsch>winget install Qalculate.Qalculate
Found Qalculate! [Qalculate.Qalculate] Version 5.3.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/Qalculate/libqalculate/releases/download/v5.3.0/qalculate-5.3.0-x64.msi
  ██████████████████████████████  60.2 MB / 60.2 MB
Successfully verified installer hash
Starting package install...
Successfully installed
```

**Reply by djmattyg007** [reply]
I'm getting the same issue with the same Qalculate upgrade.

**Reply by heksesang** [reply]
Many installers give you the option to install for all users (machine scope) or just yourself (user scope). And if the [manifest](https://github.com/microsoft/winget-pkgs/blob/master/manifests/w/WinDirStat/WinDirStat/2.0.1/WinDirStat.WinDirStat.installer.yaml) sets a different scope than the one you installed for, you'll get this error when you try to do an upgrade. I assume most manifests to be configured for machine scope if available.

I had the same issue with Sigil, in which case reinstalling it and choosing the all users option solved it.

**Comment by squeak9** [comment]
I'm also having this issue with JOSM. I'm not sure if should create an issue report, if this belongs elsewhere then let me know.

https://github.com/microsoft/winget-pkgs/tree/master/manifests/o/OpenStreetMap/Josm/

The files says this is Scope: user but when you install it the normal way using winget it says it's installed as machine. This means you can install josm using winget but you cannot upgrade it using winget.

**Steps to reproduce:**
run
```ps
winget install OpenStreetMap.Josm
```
and then:
```ps
winget list --scope machine -q josm
```
You get this:
```
Name Id                 Version Source
---------------------------------------
JOSM OpenStreetMap.Josm 19555   winget
```
But [this](https://github.com/microsoft/winget-pkgs/blob/master/manifests/o/OpenStreetMap/Josm/19555/OpenStreetMap.Josm.installer.yaml) says `Scope: user`

I could just uninstall and reinstall it every time I need to upgrade but that won't solve the problem for others.
